keps/sig-apps/3939-include-terminating-pods-as-active/README.md

```diff
@@ -0,0 +1,852 @@
+- ?
+
+### Job API Definition
+
+At the JobSpec level, we are adding a new BoolPtr field:
+
+```golang
+type JobSpec struct{
+  ...
+ // terminatingAsActive specifies if the Job controller should include terminating pods
+ // as active. If the field is true, then the Job controller will include active pods
+ // to mean running or terminating pods
+ // +optional
+ TerminatingAsActive *bool
+}
+```
+
+So we can count terminating pods separately from active or failed we need to include a new field in the JobStatus.
+
+```golang
+type JobStatus struct {
+  ...
+  // Number of terminating pods
+  // +optional
+  terminating *int32
+}
+```
+
+### Deployment/ReplicaSet API
+
+```golang
+// DeploymentSpec stores information about the strategy and rolling-update
+// behavior of a deployment.
+type DeploymentSpec struct {
+  ... 
+  // TerminatingAsActive specifies if the Deployments should include terminating pods
+ // as active. If the field is true, then the Deployment controller will include active pods
+ // to mean running or terminating pods
+ // +optional
+ TerminatingAsActive *bool
+}
+```
+
+So we can count terminating pods separately from active or failed we need to include a new field in the ReplicaSetStatus and DeploymentStatus.
+
+```golang
+type DeploymentStatus struct {
+  ...
+  // Terminating replicas states the number of replicas that are terminating
+  // +optional
+  TerminatingReplicas *int32 
+}
+```
+
+```golang
+type ReplicaSetStatus struct {
+  ...
+  // Terminating replicas states the number of replicas that are terminating
+  // +optional
+  TerminatingReplicas *int32 
+}
+```
+
+In [Option for acknowledging terminating Pods in Deployment rolling update](https://github.com/kubernetes/kubernetes/issues/107920)
+there was a request to add this as part of the `DeploymentStrategy` field.  Generally, handling terminating pods as active can be useful in both RollingUpdates and Recreating rollouts.  Having this field for both strategies allows for handling of terminating pods in both cases.  
+
+Deployments create ReplicaSets so there is a need to add a field in the ReplicaSet as well.  Since ReplicaSets are not typically
+set by users, we should add a field to the ReplicaSet that is set from the DeploymentSpec.  
+
+```golang
+// ReplicaSetSpec is the specification of a ReplicaSet.
+// As the internal representation of a ReplicaSet, it must have
+// a Template set.
+type ReplicaSetSpec struct {
+  ...
+ // TerminatingAsActive specifies if the Deployments should include terminating pods
+ // as active. If the field is true, then the Deployment controller will include active pods
+ // to mean running or terminating pods
+ // +optional
+ TerminatingAsActive *bool
+}
+```
+
+### Implementation
+
+Generally, both the Job controller and ReplicaSets utilize `FilterActivePods` in their reconciliation loop.  `FilterActivePods` gets a list of pods that are not terminating.  This KEP will include terminating pods in this list.
```

Comment:
If you have a separate field `terminating`, we don't need to change the semantics of `active`